### PR TITLE
GODRIVER-2976 Remove deprecated BSON code (phase 1)

### PR DESCRIPTION
**Pull request by qingyang-hu**
GODRIVER-2976

## Summary
Phase 1 of the deprecation to limit the files to reviewing one time.

### CODE REVIEW

File: bson/bson_test.go [hunk 1]
```diff
@@ -177,13 +178,22 @@ func TestMapCodec(t *testing.T) {
 			{"true", bsonoptions.MapCodec().SetEncodeKeysWithStringer(true), "bar"},
 			{"false", bsonoptions.MapCodec().SetEncodeKeysWithStringer(false), "foo"},
 		}
+		buf := new(bytes.Buffer)
+		enc := new(Encoder)
```
Comment (matthewdale) [inline bson/bson_test.go:182]: It's technically safe to share the buffer and `Encoder` in this case, but it creates an opportunity for unexpected interaction between test cases. We should initialize these values in the test case funcs.

File: bson/marshal_test.go [hunk 2]
```diff
@@ -26,50 +26,9 @@ import (
...
-				t.Errorf("Bytes are not equal. got %v; want %v", got, tc.want)
-				t.Errorf("Bytes:\n%v\n%v", got, tc.want)
-			}
-		})
-	}
-}
-
 func TestMarshalWithRegistry(t *testing.T) {
+	buf := new(bytes.Buffer)
+	enc := new(Encoder)
```
Comment (matthewdale) [inline bson/marshal_test.go:31]: We should initialize these values in the test case funcs.

File: bson/marshal_test.go [hunk 3]
```diff
@@ -90,6 +54,8 @@ func TestMarshalWithRegistry(t *testing.T) {
 }
 
 func TestMarshalWithContext(t *testing.T) {
+	buf := new(bytes.Buffer)
+	enc := new(Encoder)
```
Comment (matthewdale) [inline bson/marshal_test.go:58]: We should initialize these values in the test case funcs.

File: mongo/options/mongooptions.go [hunk 4]
```diff
@@ -131,12 +133,21 @@ func (af *ArrayFilters) ToArray() ([]bson.Raw, error) {
 		registry = bson.DefaultRegistry
 	}
 	filters := make([]bson.Raw, 0, len(af.Filters))
+	buf := new(bytes.Buffer)
+	enc := new(bson.Encoder)
```
Comment (matthewdale) [inline mongo/options/mongooptions.go:137]: We should initialize these values in the test case funcs.

File: mongo/options/mongooptions.go [hunk 5]
```diff
@@ -152,13 +163,22 @@ func (af *ArrayFilters) ToArrayDocument() (bson.Raw, error) {
 	}
 
 	idx, arr := bsoncore.AppendArrayStart(nil)
+	buf := new(bytes.Buffer)
+	enc := new(bson.Encoder)
```
Comment (matthewdale) [inline mongo/options/mongooptions.go:167]: We should initialize these values in the test case funcs.

File: internal/integration/unified/gridfs_bucket_operation_execution.go [hunk 6]
```diff
@@ -148,7 +148,7 @@ func executeBucketDownloadByName(ctx context.Context, operation *operation) (*op
 		case "filename":
 			filename = val.StringValue()
 		case "revision":
-			opts.SetRevision(val.AsInt32())
+			opts.SetRevision(int32(val.AsInt64()))
```
Comment (matthewdale) [inline internal/integration/unified/gridfs_bucket_operation_execution.go:151]: Check this conversion to make sure it doesn't overflow.

**Review by prestonvasquez — APPROVED** [review]
LGTM, with changes requests by @matthewdale 👍

File: mongo/options/mongooptions.go (hunk 4 above)
Comment (qingyang-hu) [reply mongo/options/mongooptions.go:137]: Will skip this as it is not in a test.

File: mongo/options/mongooptions.go (hunk 5 above)
Comment (qingyang-hu) [reply mongo/options/mongooptions.go:167]: Will skip this as it is not in a test.

**Review by prestonvasquez — APPROVED** [review]

File: mongo/options/mongooptions.go (hunk 4 above)
Comment (matthewdale) [reply mongo/options/mongooptions.go:137]: Ah, good catch. Please disregard.

File: mongo/options/mongooptions.go (hunk 5 above)
Comment (matthewdale) [reply mongo/options/mongooptions.go:167]: Ah, good catch. Please disregard.

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**